pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -389,18 +392,39 @@ func (a *Assignment) findFlavorForResourceGroup(
 			}
 		}
 
+		if shouldTryNextFlavor(representativeMode, cq.FlavorFungibility, whetherNeedBorrowing) {
+			return assignments, nil
+		}
 		if representativeMode > bestAssignmentMode {
 			bestAssignment = assignments
 			bestAssignmentMode = representativeMode
-			if bestAssignmentMode == Fit {
-				// All the resources fit in the cohort, no need to check more flavors.
-				return bestAssignment, nil
-			}
+			// if bestAssignmentMode == Fit {
+			// 	// All the resources fit in the cohort, no need to check more flavors.
+			// 	return bestAssignment, nil
+			// }
 		}
 	}
 	return bestAssignment, status
 }
 
+func shouldTryNextFlavor(representativeMode FlavorAssignmentMode, flavorFungibility v1beta1.FlavorFungibility, whetherNeedBorrowing bool) bool {
+	policyPreempt := flavorFungibility.WhenCanPreempt
+	policyBorrow := flavorFungibility.WhenCanBorrow
+	if representativeMode == Preempt && policyPreempt == v1beta1.Preempt {
```

Comment:
at this point, we don't know if preemption would be possible (there might not be enough candidates for preemption)

So how would we "try the next flavor" if preemption doesn't help.

Another thought that comes to mind is whether we should prioritize partial admission over "trying the next flavor".

I think we need more thinking about policies overall.

Reply:
Another possible implementation path is that we somehow keep some information in memory about the last flavors and modes we used. This way, in a future scheduling attempt, we can try the next flavor if preemption didn't work.

Reply:
Yes, it is better to try next flavor the next time workload been scheduled. I will try to give some implementation for this.

Reply:
Has this changed?